projects/plutus/src/current_calibration.c

```diff
@@ -0,0 +1,40 @@
+#include "current_calibration.h"
+#include "critical_section.h"
+#include "log.h"
+#include "wait.h"
+
+static void prv_callback(int32_t *value, void *context) {
+  CurrentCalibrationStorage *storage = (CurrentCalibrationStorage *)context;
+
+  storage->voltage += *value;
```

Comment:
How do you make sure this doesn't overflow?

Reply:
To prevent overflow, `CURRENT_CALIBRATION_SAMPLES` needs to be less than 16 since the adc reading is represented by a signed 28-bit number. I'll keep a comment indicating this.

Reply:
You should be checking/asserting for that. If it's a #define, use a static assert so it's detected at compile-time.

Reply:
Done